README.md

```diff
@@ -1,2 +1,331 @@
+
+
+## 2.3 Applications
+An application is a program that people use to interact with the Farcaster network. It can be standalone client that can be downloaded and run locally, or hosted software that runs on a cloud server and can be accessed through a specific client. Users are free to choose the type of application that best suits their needs. 
+
+A simple application might consist of a standalone desktop or mobile client that talks directly to a Farcaster Hub. It can publish new messages and view messages published by other accounts. Each client will need to be instantiated with a [valid signing key](#42-signer). 
+
+A more sophisticated application might introduce a proxy backend server between the Hub and the client. Advanced features like search can be implemented on this server. It might also operate as a hosted service, where it custodies keys for the user and performs all the signing on the backend.
+
+## 2.4 Hubs
+A Hub is an always-on server that validates, stores, and replicates Signed Messages. Applications store data on Hubs to make them available to their followers at all times.
+
+Hubs can fetch data from other Hubs, making it easier for mobile clients who tend to go offline a lot. For instance, Alice follows Bob and Charlie, who use separate Hubs. She configures her Hub to sync with their Hubs periodically. When her client comes online, it can fetch everyone's messages with a single call to her Hub.
+
+Every Hub monitors the Account Contract so that it can validate signed messages, and detect malicious Hubs that send forged data. Therefore, we can trust data about any user from any Hub because self-authentication prevents forgery. If Bob has a copy of Charlie's messages, Alice's server can download them and save a round trip to Charlie's Hub. Hubs can use a gossip protocol and fetch data from the closest peer with a copy instead of going to the user's actual Hub.
+
+Hubs form an **L2 network for storing social data**, though the network has different properties from blockchain-based L2s. Its consensus model has weaker consistency guarantees but stronger scalability guarantees because the network data is **shardable** down to the account level. Users can run their own Hub, share one with others or use a third-party service to host one on their behalf.
+
+# 3. Identity System
+
+The identity system ensures secure and decentralized ownership of user accounts. Users must be able to set up a new account without any third party approval. It should also be reasonably quick and cheap. For our purposes, we define this as being able to complete registration in under a minute and for less than $10. It is is also important that accounts are trustworthy and operated primarily by honest, active users. 
+
+Farcaster's Identity System is composed of two smart contracts - an **Account contract**, which creates new accounts, and a **Namespace contract** which issues usernames that can be used with accounts.
```

Comment:
Nit: these could have more descriptive names, like `AccountRegistry` and `NamespaceManager`